### PR TITLE
Added header for apps

### PR DESCRIPTION
**Pull request by AAllport**
See: https://developer.github.com/v3/apps

Fixes: #800

### CODE REVIEW

**Comment by AAllport** [comment]
Certainly!

**Comment by AAllport** [comment]
@acrobat is there a release schedule?
Unfortunately, pr on GrahamCampbell/Laravel-GitHub#95 was rejected, so would be nice to know when we can hope to see these changes hit packagist?

File: lib/Github/Api/Apps.php [hunk 1]
```diff
@@ -9,6 +9,15 @@
...
 class Apps extends AbstractApi
 {
+    use AcceptHeaderTrait;
+
+    public function configure()
+    {
+        $this->acceptHeaderValue = 'application/vnd.github.machine-man-preview+json';
+
+        return $this;
+    }
```
Comment (AAllport) [inline lib/Github/Api/Apps.php]: I'm not too sure whether this part is needed?

```suggestion
    public function configure()
    {
        $this->acceptHeaderValue = 'application/vnd.github.machine-man-preview+json';
    }
```

Comment (AAllport) [reply lib/Github/Api/Apps.php]: TIL Multiline suggestions don't work
My point, not too sure if we need the `return $this;`
Not having anything causes the test's to fail

Comment (acrobat) [reply lib/Github/Api/Apps.php]: The `return $this` is indeed not needed here, so you can remove it. Which tests fail without the `return $this`?

**Comment by acrobat** [comment]
@AAllport There is no planned release schedule, but I'm planning on publishing a new release soon as there are many fixed/features since last release!

File: lib/Github/Api/Apps.php (hunk 1 above)
Comment (acrobat) [reply lib/Github/Api/Apps.php]: I see why they will fail, just rename this method to `configurePreviewHeader`, make the method final and don't have it return anything. This should work correctly!

**Comment by AAllport** [comment]
> @AAllport There is no planned release schedule, but I'm planning on publishing a new release soon as there are many fixed/features since last release!

Looking forward to it!

**Comment by AAllport** [comment]
Well that's ruddy mysterious

**Comment by acrobat** [comment]
@AAllport the travis error is not related to these changes but something incorrectly merged on master, I've pushed a fix to the master branch. Can you rebase your branch onto master? Thanks!

File: lib/Github/Api/Apps.php [hunk 2]
```diff
@@ -9,6 +9,13 @@
  */
 class Apps extends AbstractApi
 {
+    use AcceptHeaderTrait;
+
+    final public function configurePreviewHeader()
```
Comment (acrobat) [inline lib/Github/Api/Apps.php]: Just mark it private (and remove the final), there is no need to expose this method to the public api.

**Comment by AAllport** [comment]
My git foo isn't quite as good as some be.
If I do a `git pull upstream master` and it merges, will that suffice (where upstream is this repo)?
IIRC it will merge commit. or did you specifically want a rebase?

File: lib/Github/Api/Apps.php (hunk 2 above)
Comment (AAllport) [reply lib/Github/Api/Apps.php]:
```suggestion
    private function configurePreviewHeader()
```

**Comment by acrobat** [comment]
@AAllport A merge is fine! I will squash the commits when merging the pr!

**Comment by acrobat** [comment]
Thanks a lot @AAllport! And congrats on your first contribution! 🎉

**Comment by AAllport** [comment]
For what it's worth, I believe this is my 1st functional OSS contribution, 10/10 would definitely recommend 😁
I look forward to the release
All the best!

**Comment by acrobat** [comment]
@AAllport I've published [2.12.0](https://github.com/KnpLabs/php-github-api/releases/tag/2.12.0)!

**Pull request merged**